File: SocialAppFlutter/social_app/lib/Models/members_honor.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:social_app/Models/attended_category.dart';
import 'package:social_app/Models/user.dart';

part 'json_generated_code/members_honor.g.dart';

@JsonSerializable(explicitToJson: true)
class MembersHonor {
  final User fromHonor;
  final AttendedCategory attendedCategory;

  MembersHonor(this.fromHonor, this.attendedCategory);

  factory MembersHonor.fromJson(Map<String, dynamic> json) =>
      _$MembersHonorFromJson(json);

  Map<String, dynamic> toJson() => _$MembersHonorToJson(this);
}

File: APIs/EventsService/EventsService/DTOs/MembersHonor.cs
using Newtonsoft.Json;

namespace EventsService.DTOs
{
    public class MembersHonor
    {
        [JsonProperty("fromHonor")]
        public UserObject FromHonor { get; set; } = null!;
        [JsonProperty("attendedCategory")]
        public AttendedCategory AttendedCategory { get; set; } = null!;
    }
}
